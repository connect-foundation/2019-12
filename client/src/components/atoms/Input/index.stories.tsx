import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import Input from '.';

storiesOf('0 - Atoms - Input', module)
  .add('default', () => (
    <Input
      disabled={boolean('disabled', false)}
      defaultValue={text('defaultValue', 'bookus@gmail.com')}
    />
  ))
  .add('disabled', () => (
    <Input
      disabled={boolean('disabled', true)}
      defaultValue={text('defaultValue', 'bookus@gmail.com')}
    />
  ));
