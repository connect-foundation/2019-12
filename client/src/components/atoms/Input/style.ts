import styled from 'styled-components';
import { ifProp, palette, theme } from 'styled-tools';

interface Props {
  invalid?: boolean;
}

export const Input = styled.input<Props>`
  ${theme('fontStyle.body2')}
  height: 4rem;
  width: 100%;
  padding: 0 1rem;

  color: ${palette('grayscale', 2)};
  background-color: ${ifProp(
    'disabled',
    palette('grayscale', 6),
    palette('white'),
  )};

  border: 1px solid
    ${ifProp('invalid', palette('danger'), palette('grayscale', 4))};
  border-radius: 2px;

  &:focus {
    outline: none;
  }
`;
