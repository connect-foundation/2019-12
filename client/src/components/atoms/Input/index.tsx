import React from 'react';
import * as S from './style';

interface Props {
  /** name of input */
  inputName: string;
  /** is invalid? */
  invalid?: boolean;
  /** is disabled */
  disabled?: boolean;
  /** defaultValue of input */
  defaultValue?: string;
  /** placeholder content */
  placeholder?: string;
  /** input value(state) */
  value?: string;
  /** onChange handler(setState) */
  onChange?: (e: any) => void;
}

function Input({ inputName, ...props }: Props): React.ReactElement {
  return <S.Input name={inputName} {...props} />;
}

export default Input;
