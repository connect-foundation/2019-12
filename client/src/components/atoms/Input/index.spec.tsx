import React from 'react';
import { mount } from 'enzyme';
import Input from '.';

describe('Atom / Divider', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(
      <Input
        invalid={true}
        disabled={false}
        defaultValue={'bookusgmail.com'}
        placeholder={'이메일을 입력해주세요.'}
      />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
