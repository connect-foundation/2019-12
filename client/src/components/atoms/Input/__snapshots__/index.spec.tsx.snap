// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / Divider [SNAPSHOT] 렌더링 1`] = `
.c0 {
  height: 4rem;
  width: 100%;
  padding: 0 1rem;
  border: 1px solid;
  border-radius: 2px;
}

.c0:focus {
  outline: none;
}

<Input
  defaultValue="bookusgmail.com"
  disabled={false}
  invalid={true}
  placeholder="이메일을 입력해주세요."
>
  <styled.input
    defaultValue="bookusgmail.com"
    disabled={false}
    invalid={true}
    placeholder="이메일을 입력해주세요."
  >
    <StyledComponent
      defaultValue="bookusgmail.com"
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
  height: 4rem;
  width: 100%;
  padding: 0 1rem;

  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";

  border: 1px solid
    ",
              [Function],
              ";
  border-radius: 2px;

  &:focus {
    outline: none;
  }
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      invalid={true}
      placeholder="이메일을 입력해주세요."
    >
      <input
        className="c0"
        defaultValue="bookusgmail.com"
        disabled={false}
        placeholder="이메일을 입력해주세요."
      />
    </StyledComponent>
  </styled.input>
</Input>
`;
